internal/v4/archive.go

```diff
@@ -456,27 +488,21 @@ func setArchiveCacheControl(h http.Header, idFullySpecified bool) {
 // othewise it returns nil. An error is returned if there is a problem
 // communicating with the storage.
 func (h *Handler) getPromulgatedURL(id *charm.Reference) (*charm.Reference, error) {
-	baseURL := *id
-	baseURL.Series = ""
-	baseURL.Revision = -1
-	var baseEntity mongodoc.BaseEntity
-	err := h.store.DB.BaseEntities().FindId(&baseURL).One(&baseEntity)
-	if err != nil {
-		if err == mgo.ErrNotFound {
-			// If there is no base entity then by definition it cannot be promulgated.
-			return nil, nil
-		}
+	baseEntity, err := h.store.FindBaseEntity(id, "promulgated")
+	if err != nil && errgo.Cause(err) != params.ErrNotFound {
 		return nil, errgo.Mask(err)
 	}
-	if !baseEntity.Promulgated {
+	if baseEntity == nil || !baseEntity.Promulgated {
 		return nil, nil
 	}
+	query := h.store.EntitiesQuery(&charm.Reference{
+		Series:   id.Series,
+		Name:     id.Name,
+		Revision: -1,
+	})
+	logger.Debugf("%v", query)
```

Comment:
Either delete this or make it a more informative log message (I think I'm more in favour of the former FWIW)